### PR TITLE
[OPENSSL] Upgrade to OpenSSL 1.1 and make it compilable on windows.

### PR DESCRIPTION
**Pull request by pwielders**
This prepares the OpenSSL implementation to work with the latest interface (OpenSSL1.1) it is also preparred to be compiled and used in a windows environment.

### CODE REVIEW

File: Source/cryptography/implementation/OpenSSL/Cipher.cpp [hunk 1]
```diff
@@ -111,7 +112,7 @@ class Cipher : public CipherImplementation {
         } else if (maxOutputLength < inputLength) {
             // Note: Pitfall, AES CBC/ECB will use padding
             TRACE_L1(_T("Too small output buffer, expected: %i bytes"), inputLength);
-            result = (-inputLength);
+            result = (~inputLength);
```
Comment (sebaszm) [inline Source/cryptography/implementation/OpenSSL/Cipher.cpp:115]: a) If we change this here, would be good to update the other implementations for consistency
b) this code is too lazy anyway, it needs to take the block cipher mode into account before suggesting the required size due to possible padding

**Review by sebaszm — APPROVED** [review]
looks good, the single comment I have is not critical as I don't think any client of the library actually relies on the too-small-buffer return value from cipher (except perhaps the test suite)

File: Source/cryptography/implementation/OpenSSL/Cipher.cpp (hunk 1 above)
Comment (pwielders) [reply Source/cryptography/implementation/OpenSSL/Cipher.cpp:115]: It was a warning in the windows build :-) I just made the warning go away and still achieve the same. Agree it is good to align the other implementation as well than.
I do not get point b :-)

Comment (pwielders) [reply Source/cryptography/implementation/OpenSSL/Cipher.cpp:115]: Oke, after our discussion I do get point B. Will update the pull request accordingly and indeed I stand corrected. The ~ is differetn than -. Will adapt it accordingly.

**Review by sebaszm — APPROVED** [review]
looks good!

**Pull request merged**